src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CacheAdapterPass.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Cache\Adapter\ApcuAdapter;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Christian Flothmann <christian.flothmann@xabbuh.de>
+ */
+class CacheAdapterPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->getDefinitions() as $id => $definition) {
+            $definition->setArguments($this->resolveArguments($container, $id, $definition->getArguments()));
+
+            $calls = $definition->getMethodCalls();
+
+            foreach ($calls as $c => $call) {
+                $arguments = $call[1];
+
+                foreach ($arguments as $index => $argument) {
+                    $arguments[$index] = $this->createCacheAdapter($container, $id, $argument);
+                }
+
+                $calls[$c] = array($call[0], $arguments);
+            }
+
+            $definition->setMethodCalls($calls);
+
+            $definition->setProperties($this->resolveArguments($container, $id, $definition->getProperties()));
+        }
+    }
+
+    private function resolveArguments($container, $id, array $arguments)
+    {
+        foreach ($arguments as $k => $argument) {
```

Comment:
The `$k` variable looks a bit confusing too: if it's the typical numeric index, it should be called `$i` and if it's a key index, it should be called `$key`